### PR TITLE
Montana 2022 / 2023 property tax rebates

### LINKED ISSUE #5339: Montana 2022 / 2023 property tax rebates
[Reference](https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/12/Form_2_2023_Instructions.pdf#page=5)

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #5339

### CODE REVIEW

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  2022-01-01: 500
+  2024-01-01: 675
+metadata:
+  period: year
+  unit: currency-USD
+  label: Montana property tax rebate amount
+  reference:
+    - title: 2023 Montana Form 2 Individual Income Tax Instructions, What's New?, Property Tax Rebates
+      href: https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/12/Form_2_2023_Instructions.pdf#page=5
+    - title: 2023 Montana Legislature, House Bill 222 # Legal code does not mention updated property tax rebate amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml:12]:
```suggestion
# Legal code does not mention updated property tax rebate amount
    - title: 2023 Montana Legislature, House Bill 222 
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/mt_property_tax_rebate.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    definition_period = YEAR
+    reference = "https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/12/Form_2_2023_Instructions.pdf#page=5"
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.mt.tax.income.credits.rebate.property
+        person = tax_unit.members
+        mt_property_tax = add(person, period, ["mt_property_tax"])
+        mt_property_tax_total = tax_unit.sum(mt_property_tax)
+        return min_(p.amount, mt_property_tax_total.item())
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/mt_property_tax_rebate.py:18]:
```suggestion
        return min_(p.amount, mt_property_tax_total)
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/mt/tax/income/credits/mt_property_tax_rebate.py:18]: dont think we need this

File: policyengine_us/variables/gov/states/mt/tax/income/mt_property_tax.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_property_tax.py:1]: lets just use the federal real estate taxes
'real_estate_taxes.py'

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/non_refundable.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/non_refundable.yaml:1]: are we sure it is non-refundable?

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+description: Montana provides the following property tax rebate amount.
+values:
+  2022-01-01: 675
+metadata:
+  period: year
+  unit: currency-USD
+  label: Montana property tax rebate amount
+  reference:
+    - title: 2023 Montana Form 2 Individual Income Tax Instructions, What's New?, Property Tax Rebates
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml:9]: could you add the 2022 and 2023 tax forms here? 
If the value changes between 2022 and 2023 in the tax forms, we should show that

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml (hunk 3 above)
Comment (vrathi101) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml:9]: i added 2022 property tax rebate form, i wasn't able to find one for 2023 but the form 2 instructions mention the same rate for 2022 and 2023 and online also mentions the same 675 value

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      href: https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/12/Form_2_2023_Instructions.pdf#page=5
+    - title: 2022 Montana Property Tax Rebate Claim, Form MPTR22
+      href: https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/08/2022-Montana-Property-Tax-Rebate-Form-MPTR22.pdf
+    # Legal code does not mention updated property tax rebate amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/rebate/property/amount.yaml:13]:
```suggestion
    # The Legal code provides a lower rebate amount - applying the tax form instructions value
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**